diff --git a/chrome/android/chrome_public_apk_tmpl.gni b/chrome/android/chrome_public_apk_tmpl.gni
--- a/chrome/android/chrome_public_apk_tmpl.gni
+++ b/chrome/android/chrome_public_apk_tmpl.gni
@@ -14,6 +14,7 @@ import("//chrome/android/modules/chrome_feature_modules.gni")
 import("//chrome/android/trichrome.gni")
 import("//chrome/common/features.gni")
 import("//chrome/version.gni")
+import("//brave/build/config.gni")
 import("//components/crash/android/silent_java_assert_reporting.gni")
 import("//content/public/common/features.gni")
 import("//device/vr/buildflags/buildflags.gni")
@@ -266,3 +267,6 @@ template("chrome_common_apk_or_module_tmpl") {
     if (_target_type == "android_apk") {
       command_line_flags_file = "chrome-command-line"
+      if (!defined(invoker.apk_name) && brave_android_public_apk_name != "") {
+        apk_name = brave_android_public_apk_name
+      }
     }
